types/styled-components/index.d.ts

```diff
@@ -345,8 +346,10 @@ export type ThemedCssFunction<T extends object> = BaseThemedCssFunction<
 >;
 
 // Helper type operators
-type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
-type WithOptionalTheme<P extends { theme?: T }, T> = Omit<P, "theme"> & {
+// Pick that distributes over union types
+export type PickU<T, K extends keyof T> = T extends any ? {[P in K]: T[P]} : never;
+export type OmitU<T, K extends keyof T> = T extends any ? PickU<T, Exclude<keyof T, K>> : never;
```

Comment:
Interesting that `T extends any` works around the issue 🤔

Reply:
@Jessidhia The solution came from your issue [here](https://github.com/Microsoft/TypeScript/issues/28339)

And it's explained : 
> For example, an instantiation of `T extends U ? X : Y` with the type argument `A | B | C` for `T` is resolved as `(A extends U ? X : Y) | (B extends U ? X : Y) | (C extends U ? X : Y)`.

Here : https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#distributive-conditional-types